### PR TITLE
btrfs-progs: check: report deprecated inode cache as errors

### PR DESCRIPTION
**Pull request by adam900710**
There are two reports that inode cache is causing kernel to reject those
data backrefs.

It turns out the long to be deprecated inode cache is biting us
unexpectedly.
(And the design of those special inodes has no intention to properly
fill its members like mode and transid correctly).

However the original mode btrfs check is not updated to detect them, in
fact there are special hacks to not report them as errors.
Meanwhile the lowmem mode immediately picks them as errors from day one.

The first patch fixes the original mode to report inode cache as error,
and add extra messages on how to properly fix the problem using 'btrfs
rescue clear-ino-cache'.

The second patch is a new test case for both modes of btrfs check.

### CODE REVIEW

File: check/main.c [hunk 1]
```diff
@@ -10834,6 +10838,8 @@ static int cmd_check(const struct cmd_struct *cmd, int argc, char **argv)
 
 	ret = do_check_fs_roots(&root_cache);
 	task_stop(g_task_ctx.info);
+	if (found_free_ino_cache)
+		printf("deprecated inode cache can be removed by 'btrfs rescue clear-ino-cache'\n");
```
Comment (kdave) [inline check/main.c:10842]: please use pr_verbose(LOG_DEFAULT, ...)

File: check/mode-lowmem.c [hunk 2]
```diff
@@ -5616,6 +5623,8 @@ int check_fs_roots_lowmem(void)
 
 out:
 	btrfs_release_path(&path);
+	if (found_free_ino_cache)
+		printf("deprecated inode cache can be removed by 'btrfs rescue clear-ino-cache'\n");
```
Comment (kdave) [inline check/mode-lowmem.c:5627]: pr_verbose

File: check/mode-lowmem.c [hunk 3]
```diff
@@ -2629,6 +2630,12 @@ static int check_inode_item(struct btrfs_root *root, struct btrfs_path *path)
 		return err;
 	}
 
+	if (inode_id == BTRFS_FREE_INO_OBJECTID) {
+		error("subvolume %lld has deprecated inode cache",
```
Comment (kdave) [inline check/mode-lowmem.c:2634]: If this is an error then it should also exit at some point. As it is now it looks more like a warning.

**Review by kdave — APPROVED** [review]

File: check/mode-lowmem.c (hunk 3 above)
Comment (adam900710) [reply check/mode-lowmem.c:2634]: I can change that to warning, but even if we hit an error, as long as it's not critical, we should always continue other than exiting.
Or we will not fully check the whole filesystem.

**Pull request merged**